tests/components/stream/test_worker.py

```diff
@@ -729,24 +729,31 @@ def blocking_open(stream_source, *args, **kwargs):
 
 async def test_worker_log(hass, caplog):
     """Test that the worker logs the url without username and password."""
-    stream = Stream(
-        hass,
-        "https://abcd:efgh@foo.bar",
-        {},
-        hass.data[DOMAIN][ATTR_SETTINGS],
-        dynamic_stream_settings(),
-    )
-    stream.add_provider(HLS_PROVIDER)
+    for stream_url, redacted_url in (
+        ("https://abcd:efgh@foo.bar", "https://****:****@foo.bar"),
+        (
+            "https://foo.bar/baz?user=abcd&password=efgh",
+            "https://foo.bar/baz?user=****&password=****",
+        ),
+    ):
```

Comment:
Since the logic is getting a little more complex, I suggest adding a test case with:
- url with parameters that are not redacted
- url with mix of parameters that are redacted and not redacted

(I am suggesting this because it wasn't obvious to me that the `dict.fromkeys` suggested handled these two cases since i am not familiar with what it is doing)

Reply:
Added these two cases.